### PR TITLE
Set the correct tier price for alternative website (not default) on admin order create

### PR DESCRIPTION
**Pull request by nicobatty**
### Fixed Issues (if relevant)

1. magento/magento2#18577: On sales/order_create/loadBlock/block Customer tier price remove

### Manual testing scenarios (*)

1. The reproducing scenario is specified here: https://github.com/magento/magento2/issues/18577#issuecomment-430662428
2. The changes done on the Configurable module fix the add to cart for a tier priced configurable product.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18784) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @nicobatty. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Catalog/Model/Product/Attribute/Backend/GroupPrice/AbstractGroupPrice.php [hunk 1]
```diff
@@ -331,11 +331,15 @@ private function getWebsiteId($storeId)
     /**
      * @param \Magento\Catalog\Model\Product $object
      * @param array $priceData
+     * @param int|null $storeId
      */
-    public function setPriceData($object, $priceData)
+    public function setPriceData($object, $priceData, $storeId = null)
```
Comment (sidolov) [inline app/code/Magento/Catalog/Model/Product/Attribute/Backend/GroupPrice/AbstractGroupPrice.php:336]: Unfortunately, we are not allowed to modify any public method according to our [Backward compatible development guide](https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/). Consider to define new method, fix all usages and deprecate this one
Thank you!

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @nicobatty , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**